File: 01-Sorting/ACM-Problem-Solution-And-Practice/UVA 299 - Train Swapping.c
#include<stdio.h>

int train[64], count;
int l, n, i, j, key;

int insertion_sort(int *train)
{
    /*
    *
    * Insertion sort using inner for loop
    *
    */
    for(i = 1; i < l; ++i)
    {
        key = train[i];
        for(j = i - 1; j >= 0 && train[j] > key; --j)
        {
            train[j + 1] = train[j];
            ++count;
        }
        train[j + 1] = key;
    }
  return count;
}

int main()
{
    scanf("%d", &n);
    while(n--)
    {
        count = 0;
        scanf("%d", &l);
        for(i = 0; i < l; ++i)
            scanf("%d", &train[i]);
 
        count = insertion_sort(train);
 
        printf("Optimal train swapping takes %d swaps.\n", count);
    }
    return 0;
}

File: 01-Sorting/ACM-Problem-Solution-And-Practice/UVA 299 - Train Swapping.py
def insertion_sort(train, l, count =0):
    for i in range(1, l):
        key = train[i]
        j = i-1
        while j >=0 and key < train[j]: 
            train[j+1] = train[j] 
            count = count + 1
            j -= 1
        train[j+1] = key
    return count

train = []
n = int(input())
for i in range(0,n):
    l = int(input())
    train = list(map(int, input().split()))
    count = insertion_sort(train, l)
    print("Optimal train swapping takes {} swaps.".format(count))